docs/debugging/cluster-reboot/main.go

```diff
@@ -0,0 +1,511 @@
+					continue
+				}
+
+			}
+		}
+	}
+
+	_ = os.RemoveAll(folder)
+	err = os.MkdirAll(folder, 0o777)
+	if err != nil {
+		panic(err)
+	}
+
+	var roundFile *os.File
+
+	for ri, rv := range rebootRounds {
+		for _, rv2 := range rv {
+			if rv2 != nil && len(rv2) > 0 {
+				roundFile, err = os.OpenFile(filepath.Join(folder, "round-"+strconv.Itoa(ri)), os.O_RDWR|os.O_CREATE|os.O_APPEND, 0o777)
+				if err != nil {
+					panic(err)
+				}
+				for _, rv3 := range rv2 {
+					// _, err = roundFile.WriteString(rv3.Endpoint + "\n")
+					_, err = roundFile.WriteString(strings.Replace(rv3.Endpoint, ":443", "", -1) + "\n")
+					if err != nil {
+						panic(err)
+					}
+				}
+				roundFile.Sync()
+				roundFile.Close()
+			}
+		}
+	}
+}
+
+func haveMatchingSets(s1 *Server, s2 *Server) (yes bool) {
+	for setid := range s1.Sets {
+		_, ok := s2.Sets[setid]
+		if ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+var DefaultTransport = func(secure bool) http.RoundTripper {
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:       5 * time.Second,
+			KeepAlive:     15 * time.Second,
+			FallbackDelay: 100 * time.Millisecond,
+		}).DialContext,
+		MaxIdleConns:          1024,
+		MaxIdleConnsPerHost:   1024,
+		ResponseHeaderTimeout: 60 * time.Second,
+		IdleConnTimeout:       60 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		DisableCompression:    true,
+	}
+
+	if secure {
+		tr.TLSClientConfig = &tls.Config{
+			InsecureSkipVerify: true,
+			MinVersion:         tls.VersionTLS12,
+		}
+	}
+	return tr
+}
+
+func healthCheck() {
+	defer func() {
+		r := recover()
+		if r != nil {
+			log.Println(r, string(debug.Stack()))
+		}
+	}()
+
+	hosts, err := os.ReadFile(hostfile)
+	if err != nil {
+		panic(err)
+	}
+	hostsList := bytes.Split(hosts, []byte{10})
+
+	hostMap := make(map[string]bool)
+	for _, v := range hostsList {
+		if len(v) < 1 {
+			continue
+		}
+		hostMap[string(v)] = false
+	}
+
+	defer func() {
+		fmt.Println("Post run host report...")
+		for i, v := range hostMap {
+			if v {
+				fmt.Println("healthy:", i)
+			} else {
+				fmt.Println("unhealthy:", i)
+			}
+		}
+	}()
+
+	unhealthy := 0
+	for {
+		unhealthy = 0
+		for host, healthy := range hostMap {
+			if healthy {
+				continue
+			}
+			ok, err := healthPing(host)
+			if err != nil {
+				unhealthy++
+				fmt.Println(err)
+				hostMap[host] = false
+			} else if !ok {
+				unhealthy++
+				fmt.Println("Waiting:", host)
+				hostMap[host] = false
+			} else {
+				hostMap[host] = true
+			}
+		}
+		if unhealthy == 0 {
+			return
+		}
+		fmt.Println("unhealthy hosts: ", unhealthy)
+		time.Sleep(30 * time.Second)
+	}
+}
+
+func rebootHostfile() {
+	defer func() {
+		r := recover()
+		if r != nil {
+			log.Println(r, string(debug.Stack()))
+		}
+	}()
+
+	hosts, err := os.ReadFile(hostfile)
+	if err != nil {
+		panic(err)
+	}
+	hostsList := bytes.Split(hosts, []byte{10})
```

Comment:
Just for ease of reading...
```suggestion
	hostsList := bytes.Split(hosts, []byte{'\n'})
```